src/st/cat/e_hellfire_beast.c

```diff
@@ -0,0 +1,1189 @@
+            var_s6 += 0x10;
+            if (var_s6 > 0x60) {
+                var_s6 = 0x10;
+            }
+
+            if (!(g_Timer % 2)) {
+                prevV0 = nextPrim->v0;
+                nextPrim->v0 = nextPrim->v2 = nextPrim->v1;
+                nextPrim->v1 = nextPrim->v3 = prevV0;
+            }
+            nextPrim->drawMode = DRAW_COLORS | DRAW_UNK02;
+            prim = nextPrim;
+            self->ext.hellfireBeastThorsHammer.unk88--;
+            i++;
+
+            if (i <= self->ext.hellfireBeastThorsHammer.unk8A) {
+                continue;
+            }
+
+            dx = nextPrim->x0 - self->posX.i.hi;
+            dx = abs(dx);
+            if (dx > 0x1FF) {
+                dx = 0x1FF;
+            }
+            self->hitboxOffX = -dx / 2;
+            self->hitboxWidth = dx / 2;
+            self->hitboxHeight = 8;
+            self->hitboxOffY = 8;
+
+            // FAKE! PSX requires some trickery here, was not able to resolve
+#ifdef VERSION_US
+            fake = 8;
+            while (prim != NULL) {
+                prim->drawMode = fake;
+                prim = prim->next;
+            }
+
+            if (0) {
+                while (posY)
+                    ;
+            }
+#else
+            while (prim != NULL) {
+                prim->drawMode = DRAW_HIDE;
+                prim = prim->next;
+            }
+#endif
+
+            if (!(g_Timer % 3) &&
+                self->ext.hellfireBeastThorsHammer.unk8A < 0x40) {
+                self->ext.hellfireBeastThorsHammer.unk8A++;
+            }
+            if (!(self->ext.hellfireBeastThorsHammer.castTimer & 0x1F)) {
+                g_api.PlaySfx(SFX_ELECTRICITY);
+            }
+
+            if (!--self->ext.hellfireBeastThorsHammer.castTimer) {
+                self->step++;
+            }
+            break; // exits while loop
+        }
+        break;
+    case 4:
+        DestroyEntity(self);
+        break;
+    }
+}
+
+// Related entity for the flame pillar
+void func_us_801CD614(Entity* self) {
+    if (!self->step) {
+        InitializeEntity(g_EInitInteractable);
+        self->animSet = 2;
+        self->animCurFrame = 1;
+        self->zPriority = 0x100;
+        self->drawFlags |= FLAG_DRAW_ROTATE;
+        self->flags &= ~FLAG_POS_CAMERA_LOCKED;
+        if (self->params) {
+            self->rotate = 0x800;
+            self->velocityY = FIX(1.0);
+        } else {
+            self->rotate = 0;
+            self->velocityY = FIX(-1.0);
+        }
+    }
+    self->posY.val += self->velocityY;
+    self->poseTimer++;
+    if (!(self->poseTimer % 2)) {
+        self->animCurFrame += 1;
+    }
+    if (self->poseTimer > 0xF) {
+        DestroyEntity(self);
+    }
+
+#ifdef VERSION_PSP
+    if (self->ext.hellfireBeastThorsHammer.entity != NULL &&
+        !self->ext.hellfireBeastThorsHammer.entity->entityId) {
+#else
+    if (!self->ext.hellfireBeastThorsHammer.entity->entityId) {
+#endif
```

Comment:
This can probably be split up into two if statements for psp, not sure.
That would let you ifdef only the first psp if statement, but you would need to ifdef a closing brace so I'm not sure which is the better choice for readability.

Reply:
Yeah I considered going down that path but went with this for the reason you mentioned, I think it is slightly easier to read with just one conditional